src/dataflow/src/source/timestamp.rs

```diff
@@ -232,6 +399,13 @@ impl TimestampBindingRc {
     /// frontier. The source can be correctly replayed from any `as_of` in advance of
     /// the compaction frontier after this operation.
     pub fn set_compaction_frontier(&mut self, new_frontier: AntichainRef<Timestamp>) {
+        assert!(
+            self.compaction_frontier.borrow().is_empty()
```

Comment:
I've removed the assert for now but yeah the since frontier starts as empty, and in that case the `less_equal` implementation doesn't return true (ie `less_equal([], [1])` is false because technically theres nothing in `[]` that is less than or equal to something in `[1]`)

In differential my understanding is that a frontier of `[]` can be used to indicate "this frontier has not been initialized yet" or also "this frontier is after all possible times" (this interpretation makes more sense if you think of the frontier as "the set of times bigger than the ones I know about")

I think you could change this protocol to always have the frontier initialized to `[0]` so the uninitialized case doesn't happen but I wasn't 100% sure it was correct

Reply:
> In differential my understanding is that a frontier of [] can be used to indicate "this frontier has not been initialized yet" or also "this frontier is after all possible times" (this interpretation makes more sense if you think of the frontier as "the set of times bigger than the ones I know about")

Whoa! Wait. A frontier of `[]` means "no times at all", in that it describes the set of times greater or equal to some element of the frontier, but with no elements there are none. A frontier of `[]` conventionally describes a completed time interval (it is the last frontier you can have in an increasing sequence of them).

Reply:
I think that understanding was due to a conversation where I said that the reason the TODO at https://github.com/MaterializeInc/materialize/blob/bd02136463ee5dce907165bd88005eb07088960d/src/ore/src/antichain.rs#L59 was difficult is because the initial empty case was not differentiable from the end case. That code might be busted, or it's a special thing about AntichainTokens? Unclear to me.

Reply:
Ah, maybe we should peek at that too. I would start out a fresh `AntichainToken` thing at the singleton set containing `T::minimum()`, which is the smallest frontier one can have.

Reply:
I addressed this issue in #6864